tests/run/carray_slicing.pyx

```diff
@@ -275,3 +277,214 @@ def struct_ptr_iter():
+# tests for coercing slice of c-array to python list
+
+def test_coercing_c_slice_to_py_start_end():
+    """
+    >>> test_coercing_c_slice_to_py_start_end()
+    [4, 9]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[2:4]
+
+def test_coercing_c_slice_to_py__end():
+    """
+    >>> test_coercing_c_slice_to_py__end()
+    [0, 1]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[:2]
+
+
+def test_coercing_c_slice_to_py_from_():
+    """
+    >>> test_coercing_c_slice_to_py_from_()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+cpdef list test_coercing_c_slice_in_cpedef_list():
+    """
+    >>> test_coercing_c_slice_in_cpedef_list()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+def test_c_to_py_coercing_array_in_var():
+    """
+    >>> test_c_to_py_coercing_array_in_var()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    b = 5555
+    for i in range(10):
+        a[i] = i**2
+    b = a[8:]
+    return b
+
+cpdef list test_c_to_py_coercing_array_in_var_2():
+    """
+    >>> test_c_to_py_coercing_array_in_var_2()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    b = 5555
+    for i in range(10):
+        a[i] = i**2
+    b = a[8:]
+    return b
+
+
+cpdef tuple test_coercing_c_slice_to_tuple():
+    """
+    >>> test_coercing_c_slice_to_tuple()
+    (64, 81)
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+cpdef set test_coercing_c_slice_to_set():
+    """
+    >>> test_coercing_c_slice_to_set()
+    set([64, 81])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+cpdef frozenset test_coercing_c_slice_to_frozenset():
+    """
+    >>> test_coercing_c_slice_to_frozenset()
+    frozenset([64, 81])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+def test_coercing_c_slice_by_casting():
+    """
+    >>> test_coercing_c_slice_by_casting()
+    [25, 36, 49, 64, 81]
+    (25, 36, 49, 64, 81)
+    set([64, 25, 36, 81, 49])
+    frozenset([64, 25, 36, 81, 49])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+
+    print(<list> a[5:])
+    print(<tuple> a[5:])
+    print(<set> a[5:])
+    print(<frozenset> a[5:])
+
+
+
+
+
+# Error !!!!!!!!
+#def test_coercing_c_slice_by_casting():
+#    """
+#    >>> test_coercing_c_slice_by_casting()
+#    [25, 36, 49, 64, 81]
+#    (25, 36, 49, 64, 81)
+#    set([64, 25, 36, 81, 49])
+#    frozenset([64, 25, 36, 81, 49])
+#    """
+#    cdef int i
+#    cdef int a[10]
+#    b = 5555
+#    for i in range(10):
+#        a[i] = i**2
+#    b = a
```

Comment:
seems like bug for coercion c-array to py-types (similar occurs for slice)
Do I have to do fixes in this pull request?